### PR TITLE
Add CommitHandler::associated_check_runs

### PR DESCRIPTION
**Pull request by iamjpotts**
The crate currently supports getting the check runs of a check suite.

This adds support for getting the check runs of a specific commit, such as the last commit of a branch related to a pull request, or the last commit in the default branch.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ## [0.29.3](https://github.com/XAMPPRocky/octocrab/compare/v0.29.2...v0.29.3) - 2023-08-15
+- add associated_check_runs to CommitHandler
```
Comment (XAMPPRocky) [inline CHANGELOG.md:11]: You don't need to add anything to CHANGELOG, we have an automation that does that for us.

**Comment by XAMPPRocky** [comment]
Thank you for your PR! LGTM, I would just remove the CHANGELOG entry and then we can merge.

File: CHANGELOG.md (hunk 1 above)
Comment (iamjpotts) [reply CHANGELOG.md]: Reverted this file.

**Comment by XAMPPRocky** [comment]
Thank you for your PR, and congrats on your first contribution! 🎉

**Pull request merged**